hack/test-cmd.sh

```diff
@@ -257,6 +251,12 @@ echo "config files: ok"
 export KUBECONFIG="${HOME}/.kube/non-default-config"
 export CLUSTER_ADMIN_CONTEXT=$(oc config view --flatten -o template -t '{{index . "current-context"}}')
 
+pushd "$OS_ROOT" > /dev/null
+restore_nullglob=$(shopt -p nullglob)
```

Comment:
I would never in a million years understand what this bash did.  That
worries me a bit, because it means we can't debug a problem in this area
and instead have to go find a bash expert the next time it's broken.  I'd
rather be a bit less tricky even if it's more lines of code (the find
snippet for instance is instantly debuggable by even bash novices).

On Oct 5, 2015, at 10:43 AM, David Eads <notifications@github.com> wrote:

In hack/test-cmd.sh
<https://github.com/openshift/origin/pull/4930#discussion_r41151610>:

> @@ -257,6 +251,12 @@ echo "config files: ok"
>  export KUBECONFIG="${HOME}/.kube/non-default-config"
>  export CLUSTER_ADMIN_CONTEXT=$(oc config view --flatten -o template -t '{{index . "current-context"}}')
>
> +pushd "$OS_ROOT" > /dev/null
> +restore_nullglob=$(shopt -p nullglob)

which is the weird default behaviour

Yeah, I hate that weird behavior. I mean, I totally knew it did that. :)

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4930/files#r41151610>.